python/pyarrow/tests/test_array.py

```diff
@@ -2615,6 +2615,8 @@ def __arrow_array__(self, type=None):
     expected = pa.array([1, 2, 3], type=pa.float64())
     assert result.equals(expected)
 
+    assert (result == None) is False
```

Comment:
I think this test is otherwise unrelated, so I would rather add this to `test_array_eq` (or maybe as a new test adjacent to that one). At once, you could maybe also add a test for other scalars like `result == 1` (that will also always return False)

Reply:
👍